### PR TITLE
Bjorncs/remove deprecated authorization

### PR DESCRIPTION
**Pull request by bjorncs**
This PR removes some of the old deployment authorization logic and replaces it with a simplified set of rules.  Please give extra attention to the updated `DeployAuthorizer.throwIfUnauthorizedForDeploy` method.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java [hunk 1]
```diff
@@ -43,36 +39,36 @@ public void throwIfUnauthorizedForDeploy(Principal principal,
...
-        if (! isScrewdriverPrincipal(principal))
-            throw loggedForbiddenException(
-                    "Principal '%s' is not a screwdriver principal, and does not have deploy access to application '%s'",
-                    principal.getName(), applicationId.toShortString());
+        if (!environmentRequiresAuthorization(environment))  return;
 
-        return tenant.isAthensTenant();
-    }
+        if (principal == null) {
+            throw loggedUnauthorizedException("Principal '%s' is not authenticated.", principal.getName());
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: Won't `principal.getName()` throw a NPE here? Consider adding test that hits this path.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java [hunk 2]
```diff
@@ -43,36 +39,36 @@ public void throwIfUnauthorizedForDeploy(Principal principal,
...
-    private void checkAthenzCredentials(Principal principal, Tenant tenant, ApplicationId applicationId) {
-        AthenzDomain domain = tenant.getAthensDomain().get();
-        if (! (principal instanceof AthenzPrincipal))
-            throw loggedForbiddenException("Principal '%s' is not authenticated.", principal.getName());
+        AthenzPrincipal athenzPrincipal = (AthenzPrincipal) principal;
+        AthenzDomain principalDomain = athenzPrincipal.getDomain();
 
-        AthenzPrincipal athensPrincipal = (AthenzPrincipal)principal;
-        if ( ! hasDeployAccessToAthenzApplication(athensPrincipal, domain, applicationId))
+        if (!principalDomain.equals(AthenzUtils.SCREWDRIVER_DOMAIN))
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: There seems to be a mix of braces/no braces for single statement blocks in this method now. I suggest just using braces for all cases.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java (hunk 1 above)
Comment (bjorncs) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: Will fix.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java (hunk 2 above)
Comment (bjorncs) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: I will add curly braces on all statement to be consistent.

**Comment by bjorncs** [comment]
@mpolden PTAL

**Comment by mpolden** [comment]
LGTM. Merge at will after freeze is lifted.

**Comment by bjorncs** [comment]
Build fails on unrelated test error in _configserver_. Merging.

**Pull request merged**